packages/react-native/Libraries/Text/Text/RCTTextView.mm

```diff
@@ -428,6 +445,105 @@ - (NSView *)hitTest:(NSPoint)point
   return isTextViewClick ? self : hitView;
 }
 
+- (NSNumber *)reactTagAtMouseLocationFromEvent:(NSEvent *)event
+{
+  NSPoint locationInSelf = [self convertPoint:event.locationInWindow fromView:nil];
+  NSPoint locationInInnerTextView = [self convertPoint:locationInSelf toView:_textView]; // This is needed if the parent <Text> view has padding
+  return [self reactTagAtPoint:locationInInnerTextView];
+}
+
+- (void)mouseEntered:(NSEvent *)event
+{
+  // superclass invokes self.onMouseEnter, so do this first
+  [super mouseEntered:event];
+
+  [self updateHoveredSubviewWithEvent:event];
+}
+
+- (void)mouseExited:(NSEvent *)event
+{
+  [self updateHoveredSubviewWithEvent:event];
+
+  // superclass invokes self.onMouseLeave, so do this last
+  [super mouseExited:event];
+}
+
+- (void)mouseMoved:(NSEvent *)event
+{
+  [super mouseMoved:event];
+  [self updateHoveredSubviewWithEvent:event];
+}
+
+- (void)updateHoveredSubviewWithEvent:(NSEvent *)event
+{
+  RCTUIView *hoveredView = nil;
+
+  if ([event type] != NSEventTypeMouseExited && _descendantViews != nil) {
+    NSNumber *reactTagOfHoveredView = [self reactTagAtMouseLocationFromEvent:event];
+
+    if ([reactTagOfHoveredView isEqualToNumber:self.reactTag]) {
+      // We're hovering over the root Text element
+      hoveredView = self;
+    } else {
+      // Maybe we're hovering over a child Text element?
+      NSUInteger index = [_descendantViews indexOfObjectPassingTest:^BOOL(RCTUIView * _Nonnull view, NSUInteger idx, BOOL * _Nonnull stop) {
+        *stop = [[view reactTag] isEqualToNumber:reactTagOfHoveredView];
+        return *stop;
+      }];
+      if (index != NSNotFound) {
+        hoveredView = _descendantViews[index];
+      }
+    }
+  }
+
+  if (_currentHoveredSubview == hoveredView) {
```

Comment:
I think _technically_ this is too early of a return if the layering of views changes underneath the hovered view changed, but that seems an exceptionally exotic case and I wouldn't worry about it without a good issue and repro.

Reply:
I'm not sure if this would ever be the case. Either way, if these are the same view, then neither of the loops below will do anything since we'll immediately fail the `view != commonAncestor` test.